screens/locationData.js

```diff
@@ -0,0 +1,137 @@
+import React, { Component } from 'react';
+import { View, Text, TouchableOpacity, StyleSheet, FlatList } from 'react-native';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import I18n from '../assets/i18n/i18n';
+import Icon from '../components/Icon';
+import colours from '../styles/colours';
+import fonts from '../styles/fonts';
+import { pushLocalLocations } from '../actions/locations';
+import { requestPushPermission } from '../actions/permissions';
+import { showPushDialog } from '../actions/settings';
+import FailedListItem from '../components/FailedListItem';
+import { getStats, getFailedLocations } from '../selectors/stats';
+
+const styles = StyleSheet.create({
+  container: {
+    width: '100%',
+    height: '100%',
+    flexDirection: 'column',
+  },
+  button: {
+    alignSelf: 'flex-start',
+    borderWidth: 1,
+    borderRadius: 5,
+    borderColor: colours.black,
+    backgroundColor: colours.greys.lightest,
+    padding: 10,
+    marginHorizontal: 5,
+    marginVertical: 10,
+  },
+  first: {
+    borderTopWidth: 1,
+  },
+  section: {
+    flexDirection: 'row',
+    paddingVertical: 15,
+    paddingHorizontal: 10,
+    backgroundColor: colours.greys.lightest,
+    borderBottomWidth: 1,
+    borderColor: colours.black,
+    alignItems: 'center',
+  },
+  sectionText: {
+    fontSize: fonts.large,
+  },
+});
+
+class LocationData extends Component {
+  state = {
+    numberUploading: 0,
+  }
+
+  uploadLocations = () => {
+    this.setState({ numberUploading: this.props.stats.offline + this.props.stats.failed });
```

Comment:
There are a whole bunch of fun issues around setting state based on props. It is better to use props directly if you can. Check out this article for context: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html